anvio/kegg.py

```diff
@@ -5179,35 +5179,19 @@ def run_enrichment_stats(self):
 
         self.progress.update('Converting modules mode output into input for enrichment script')
         enrichment_input_path = filesnpaths.get_temp_file_path()
+
         if anvio.DEBUG:
             self.progress.reset()
             self.run.info("Temporary input file for enrichment script", enrichment_input_path)
-        self.get_enrichment_input(enrichment_input_path)
 
-        cmd = f"anvi-script-enrichment-stats --input {enrichment_input_path} --output {self.output_file_path}"
-        log_file_path = filesnpaths.get_temp_file_path()
+        self.get_enrichment_input(enrichment_input_path)
 
-        self.progress.update("Running Amy's enrichment")
-        utils.run_command(cmd, log_file_path)
         self.progress.end()
-        if not filesnpaths.is_file_exists(self.output_file_path, dont_raise=True):
-            raise ConfigError("It looks like something went wrong during the enrichment analysis. "
-                              f"We don't know what happened, but this log file could contain some clues: {log_file_path}")
-
-        if filesnpaths.is_file_empty(self.output_file_path):
-            raise ConfigError("It looks like something went wrong during the functional enrichment analysis. "
-                              "An output file was created, but it is empty "
-                              f"We don't know why this happened, but this log file could contain some clues: {log_file_path}")
 
+        # run the enrichment analysis
+        enrichment_stats = utils.run_functional_enrichment_stats(enrichment_input_path,
+                                                                 self.output_file_path,
+                                                                 run=self.run,
+                                                                 progress=self.progress)
```

Comment:
I haven't tested this, @ivagljiva, would you mind making sure the output file is identical to the one you have tested previously?

Reply:
Sorry for the delay, @meren. I just tested it, and the output file is identical to my previous copy (when they are sorted, which is as expected). :)